roles/satellite-clone/tasks/main.yml

```diff
@@ -250,16 +259,8 @@
   - name: Wait for foreman-tasks service to start
     command: sleep 300
 
-  - name: Get candlepin credentials
-    command: sed -n -e 's/^.*myDS.password=//p' /etc/candlepin/candlepin.conf
-    register: candlepin_password
-
-  - set_fact:
-      clone_candlepin_password: "{{ candlepin_password.stdout }}"
-      cacheable: true
-
   - name: Migrate candlepin db
-    command: "/usr/share/candlepin/cpdb --update -p {{ clone_candlepin_password }}"
+    command: "/usr/share/candlepin/cpdb --update --dbhost={{ candlepin_db_host.value }} --password={{ candlepin_db_password.value }}"
```

Comment:
Not sure if we should pass more options here. On one hand, we've only passed the password for a while and haven't seen any issues, on the other hand foreman-installer passes more options https://github.com/theforeman/puppet-candlepin/blob/f32c2e7d9c7326a24a5795714a57bbc18c1e8e33/lib/puppet/provider/cpdb_update/cpdb_update.rb#L21-L28

This cpdb step was originally done to makes sure the backup is brought up to the latest z-version of satellite that is installed, for instance if the backup is 6.6.1 and the target server's sat version is 6.6.3, we want any changes in the candlepin db to be applied. This is likely being done in satellite-installer --upgrade step now, which we have for the same reason. 

I'm not sure if there was a reason the cpdb step was added before that step though, so would like to check a few things and confirm my assumptions before removing or gating to older versions.

Reply:
+1 to removing or gating as appropriate. It certainly gives the appearance of not being necessary.

Reply:
I'm not too clear on which versions of the installer are doing what, here are two commits in that area:
https://github.com/theforeman/foreman-installer/commit/286456500c911f3aa997b8b16cb597e67cbd2ba6#diff-b8060b8893bfb1a23cadc8c756
https://github.com/theforeman/puppet-candlepin/commit/fadb503a9c7fc12ac221f1dd74c0792540c33750

I can look into it more, but will keep this step for now.